src/routes/api/v1/search-index/news-and-events/+server.ts

```diff
@@ -0,0 +1,115 @@
+import { json, error } from "@sveltejs/kit";
+import algoliasearch from "algoliasearch";
+import { PUBLIC_ALGOLIA_APP_ID, PUBLIC_ALGOLIA_INDEX } from "$env/static/public";
+import { ALGOLIA_API_KEY } from "$env/static/private";
+import { authenticateRequest } from "$lib/services/server";
+import type { AlgoliaMetadataRecord } from "../types";
+import { format, parseISO } from "date-fns";
+
+const algoliaClient = algoliasearch(PUBLIC_ALGOLIA_APP_ID, ALGOLIA_API_KEY);
+const index = algoliaClient.initIndex(PUBLIC_ALGOLIA_INDEX);
+
+const CONTENTFUL_ACTIONS = {
+  PUBLISH: "ContentManagement.Entry.publish",
+  UNPUBLISH: "ContentManagement.Entry.unpublish",
+  DELETE: "ContentManagement.Entry.delete",
+};
+
+const addNewsMetadata = (algoliaIndexObject: AlgoliaMetadataRecord) => {
+  // Add some sensible defaults in if metaTitle and metaDescription are undefined.
+  // `title` is required for news content in Contentful, so that's an easy fallback here
+  if (algoliaIndexObject.metaTitle === undefined) {
+    algoliaIndexObject.metaTitle = algoliaIndexObject.title;
+  }
+
+  if (algoliaIndexObject.metaDescription === undefined) {
+    const { subhead, type, publicationDate } = algoliaIndexObject;
+
+    if (subhead) {
+      algoliaIndexObject.metaDescription = subhead;
+    } else if (publicationDate) {
+      const date = parseISO(publicationDate);
+      algoliaIndexObject.metaDescription = `${type} published on ${format(date, "MMMM d, yyyy")}`;
+    }
+  }
+};
+
+const addEventMetadata = (algoliaIndexObject: AlgoliaMetadataRecord) => {
+  if (algoliaIndexObject.metaTitle === undefined) {
+    algoliaIndexObject.metaTitle = algoliaIndexObject.shortTitle;
+  }
+
+  if (algoliaIndexObject.metaDescription === undefined) {
+    const { eventSummary, eventDateAndTime } = algoliaIndexObject;
+    if (eventSummary) {
+      algoliaIndexObject.metaDescription = eventSummary;
+    } else if (eventDateAndTime) {
+      const date = parseISO(eventDateAndTime);
+      algoliaIndexObject.metaDescription = `Event on ${format(date, "MMMM d, yyyy")}`;
+    }
+  }
+};
```

Comment:
this is basically fleshing out what `getMetaTitleAndDescription`, but without the side effects that was having on `pageMetadata` entries